cdap-ui/app/directives/timeline/timeline-ctrl.js

```diff
@@ -138,32 +184,54 @@ function TimelineController ($scope, LogViewerStore, LOGVIEWERSTORE_ACTIONS, myL
 
   screenSize = LogViewerStore.getState().fullScreen;
 
-  if (!this.namespaceId || !this.appId || !this.programType || !this.programId || !this.runId) {
+  if (!(this.namespaceId && this.previewId) && !(this.namespaceId && this.appId && this.programType && this.programId && this.runId)) {
```

Comment:
I'm not exactly sure what is going on here. If I am not wrong are we checking if group of these values are present?

If yes can we put it in a variable such that it is simpler to read?

```
let validPreview = this.namespaceId && this.previewId;
let validProgram = this.namespaceId && this.appId && this.programType && this.programId && this.runId;

if (!validPreview && !validProgram) {
  ... 
 }
```